File: ReverseString.cpp
class Solution {
public:
    string reverseString(string s) {
        for (int i = 0; i < s.size()/2; i++) {
            char tmp = s[s.size() - i - 1];
            s[s.size() - i - 1] = s[i];
            s[i] = tmp;
        }
        return s;
    }
};

//1Y, 12 ms, beats 28.49% of cppsubmissions, slower than java

File: ReverseString.java
public class Solution {
    public String reverseString(String s) {
        if(s == null || s.equals("")) return s;
        char[] arrChar = s.toCharArray();
        for (int i = 0, j = arrChar.length-1; i <= j; i++, j--) {
            char temp = arrChar[i];
            arrChar[i] = arrChar[j];
            arrChar[j] = temp;
        }
        return new String(arrChar);
    }
}

//copy from Shans.Xia. it spends 2ms, beats 90.17% of javasubmissions. surprising Java 0_o